aspnet/migration/aspnetcoremigration.rst

```diff
@@ -0,0 +1,190 @@
+Migrating from ASP.NET to ASP.NET Core
+======================================
+
+In this article:
+
+.. contents:: Sections
+  :local:
+  :depth: 1  
+
+Overview
+--------
+
+This migration guide will cover the essentials on how to migrate projects from ASP.NET to ASP.NET Core. 
+
+For information on migrating from DNX to .NET CLI, see the `DNX RC1->RC2 migration guide <http://placeholder>`_. 
+
+For information regarding a complete list of breaking changes in RC2 for ASP.NET, see the `ASP.NET Announcements page <https://github.com/aspnet/announcements/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0.0-rc2>`_
+
+Implement main with WebHostBuilder logic
+----------------------------------------
+
+Since ASP.NET core apps are just console apps, you must explicitly call ``Main`` to build up the host and tell it to start listening.
+
+.. code-block:: c#
+
+  public class Program
+  {
+    public static void Main(string[] args) 
+      => WebApplication.Run<Startup>(args);
+  }
+
+Configure the webroot and port in the ``hosting.json`` file.
+webroot definition moved from project.json to hosting.json
+
+WebHostBuilder API updates
+--------------------------
+
+All classes prefixed with WebApplication have been renamed to WebHost. This includes:
+
+===========================    =========================
+IWebApplicationBuilder         IWebHostBuilder
+WebApplicationBuilder          WebHostBuilder
+IWebApplication                IWebHost
+WebApplication                 WebHost
+WebApplicationOptions          WebHostOptions
+WebApplicationDefaults         WebHostDefaults
+WebApplicationService          WebHostService
+WebApplicationConfiguration    WebHostConfiguration
+===========================    =========================
+
+In the ``project.json`` file, ``web`` no longer exists as a command in the ``commands`` section. Use ``run`` or ``dotnet <dllname>`` instead  
+
+.. code-block:: JavaScript
+
+  "commands": {
+    "web": "Microsoft.AspNet.Server.Kestrel",
+    "ef": "EntityFramework.Commands"
+  },
+
+IHostingEnvironment changes 
+---------------------------
+
+All environment variables are now prefixed with the ``ASPNETCORE_`` prefix.
+
+======================  =========================    
+Old prefix              New prefix                           
+======================  =========================  
+ASPNET_WEBROOT          ASPNETCORE_WEBROOT
+ASPNET_SERVER           ASPNETCORE_SERVER
+ASPNET_APP              ASPNETCORE_APP
+ASPNET_ENVIRONMENT      ASPNETCORE_ENVIRONMENT
+ASPNET_DETAILEDERRORS   ASPNETCORE_DETAILEDERRORS
+======================  =========================  
+
+The hosting configuration keys are now consistent with the command line, environment variables, and ``hosting.json`` values. The ``Microsoft.AspNet.Hosting.json`` configuration file was renamed to ``hosting.json``.
+
+Namespace and package name changes
```

Comment:
I'd like to see this section closer to the top with a little intro text stating that ASP.NET 5 was renamed to ASP.NET Core 1.0.